OptimizelySDK/Optimizely.cs

```diff
@@ -1093,13 +1352,54 @@ public OptimizelyConfig GetOptimizelyConfig()
                 return ((IOptimizelyConfigManager)ProjectConfigManager).GetOptimizelyConfig();
             }
 
-            Logger.Log(LogLevel.DEBUG, "ProjectConfigManager is not instance of IOptimizelyConfigManager, generating new OptimizelyConfigObject as a fallback");
+            Logger.Log(LogLevel.DEBUG,
+                "ProjectConfigManager is not instance of IOptimizelyConfigManager, generating new OptimizelyConfigObject as a fallback");
 
             return new OptimizelyConfigService(config).GetOptimizelyConfig();
         }
 
         #endregion FeatureFlag APIs
 
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="segmentOptions">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        public string[] FetchQualifiedSegments(string userId, List<OdpSegmentOption> segmentOptions)
```

Comment:
This should not be public. It'll be called from OptimizelyUserContext only. "internal"?